diff -uraN libratbag-0.15/ratbagd/ratbagd-json.c omv-libratbag-0.15/ratbagd/ratbagd-json.c
--- libratbag-0.15/ratbagd/ratbagd-json.c	2020-09-25 22:38:01.000000000 +0200
+++ omv-libratbag-0.15/ratbagd/ratbagd-json.c	2021-03-31 20:32:21.103679046 +0200
@@ -515,6 +515,7 @@
 	JsonArray *arr;
 	int r = -EINVAL;
 	g_autoptr(GError) error = NULL;
+    g_autoptr(GList) list = NULL;
 
 	error = 0;
 	parser = json_parser_new();
@@ -540,7 +541,7 @@
 	num_buttons = device->num_buttons;
 	num_leds = device->num_leds;
 
-	g_autoptr(GList) list = json_array_get_elements(arr);
+	list = json_array_get_elements(arr);
 	GList *l = list;
 	int idx = 0;
 	while (l != NULL) {
